community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -1011,7 +1049,11 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
     private var _next: NodeValue = fetchNext()
 
     private def fetchNext() = {
-      if (nodeCursor.next()) fromNodeProxy(entityAccessor.newNodeProxy(nodeCursor.nodeReference()))
+      if (nodeCursor.next()) {
+        val nodeProxy = entityAccessor.newNodeProxy(nodeCursor.nodeReference())
+        val secureNodeProxy = entityAccessor.newControlledNodeProxy(nodeProxy, transactionalContext.securityContext.mode())
```

Comment:
Since this is executed a lot (on all node scan) I think using inheritance instead of composition for ControlledNodeProxy may be warranted for performance reasons.